src/interface/pipe.rs

```diff
@@ -89,60 +168,127 @@ impl EmulatedPipe {
             return false;
         }
     }
+
+    /// # Description
+    /// Checks if pipe is currently ready for writing, used by select/poll etc
+    ///
+    /// # Returns
+    ///
+    /// True if descriptor is ready for writing, false if it will block
+    ///
     pub fn check_select_write(&self) -> bool {
         let write_end = self.write_end.lock();
         let pipe_space = write_end.remaining();
 
-        return pipe_space != 0;
+        // Linux considers a pipe writeable if there is at least PAGE_SIZE (PIPE_BUF) remaining space (4096 bytes)
+        return (self.size - pipe_space) > PAGE_SIZE;
     }
 
-    // Write length bytes from pointer into pipe
+    /// ### Description
+    ///
+    /// write_to_pipe writes a specified number of bytes starting at the given pointer to a circular buffer.
+    ///
+    /// ### Arguments
+    ///
+    /// write_to_pipe accepts three arguments:
+    /// * `ptr` - a pointer to the data being written.
+    /// * `length` - the amount of bytes to attempt to write
+    /// * `nonblocking` - if this attempt to write is nonblocking
+    ///
+    /// ### Returns
+    ///
+    /// Upon successful completion, the amount of bytes written is returned.
+    /// In case of a failure, an error is returned to the calling syscall.
```

Comment:
What errors?  Please list and explain when...

Reply:
I dont understand what you mean? Don't I list the errors directly below?